packages/fx-core/src/plugins/resource/frontend/plugin.ts

```diff
@@ -169,23 +170,33 @@ export class FrontendPluginImpl {
     let runtimeEnv: RuntimeEnvironment | undefined;
     let aadEnv: AADEnvironment | undefined;
 
-    const functionPlugin = ctx.configOfOtherPlugins.get(DependentPluginInfo.FunctionPluginName);
-    if (functionPlugin) {
-      functionEnv = {
-        defaultName: functionPlugin.get(DependentPluginInfo.FunctionDefaultName) as string,
-        endpoint: functionPlugin.get(DependentPluginInfo.FunctionEndpoint) as string,
-      };
-    }
-
     if (isArmSupportEnabled()) {
-      const endpoint = getArmOutput(ctx, ArmOutput.SimpleAuthEndpoint) as string;
-      if (endpoint) {
+      const functionPlugin = ctx.configOfOtherPlugins.get(DependentPluginInfo.FunctionPluginName);
+      if (functionPlugin) {
+        const functionEndpoint = getArmOutput(ctx, FunctionArmOutput.Endpoint) as string;
+        functionEnv = {
+          defaultName: functionPlugin.get(DependentPluginInfo.FunctionDefaultName) as string,
```

Comment:
@dooriya is working on moving the `defaultFunctionName` from function config to project setting. Please decide which one we use in this PR. https://github.com/OfficeDev/TeamsFx/pull/1944

Reply:
merged the update of defaultFunctionName. Thanks.